src/ts/proposal.ts

```diff
@@ -107,15 +120,70 @@ export async function generateProposal(
       safeDeploymentAddresses,
       ethers,
     );
+  const cowTokenContract = await ethers.getContractAt(
+    "CowProtocolToken",
+    cowToken,
+  );
+  const multiTokenMediator = await ethers.getContractAt(
+    "IOmnibridge",
+    settings.bridge.multiTokenMediatorETH,
+  );
+  const approvalOmniBridgeTx = {
+    to: cowToken,
+    value: 0,
+    data: cowTokenContract.interface.encodeFunctionData("approve", [
+      settings.bridge.multiTokenMediatorETH,
+      settings.bridge.amountToRelay,
+    ]),
+    operation: 0,
+  };
+
+  const relayTestFundsToOmniBridgeTx = {
+    to: settings.bridge.multiTokenMediatorETH,
+    value: 0,
+    data: multiTokenMediator.interface.encodeFunctionData("relayTokens", [
+      cowToken,
+      cowDao,
+      settings.bridge.amountToRelay,
+    ]),
+    operation: 0,
+  };
+
+  const transferCowTokenToCowDao = {
+    to: cowToken,
+    value: 0,
+    data: cowTokenContract.interface.encodeFunctionData("transfer", [
+      cowDao,
+      totalSupply.sub(settings.bridge.amountToRelay),
+    ]),
+    operation: 0,
+  };
 
   return {
-    steps: [
-      cowDaoCreationTransaction,
-      teamControllerCreationTransaction,
-      investorFundsTargetCreationTransaction,
-      cowTokenCreationTransaction,
-      virtualCowTokenCreationTransaction,
-    ].map((step) => ({ ...step, value: step.value.toString() })),
+    steps: {
+      cowDaoCreationTransaction: transformMetaTransaction(
+        cowDaoCreationTransaction,
+      ),
+      teamControllerCreationTransaction: transformMetaTransaction(
+        teamControllerCreationTransaction,
+      ),
+      investorFundsTargetCreationTransaction: transformMetaTransaction(
+        investorFundsTargetCreationTransaction,
+      ),
+      cowTokenCreationTransaction: transformMetaTransaction(
+        cowTokenCreationTransaction,
+      ),
+      virtualCowTokenCreationTransaction: transformMetaTransaction(
+        virtualCowTokenCreationTransaction,
+      ),
+      approvalOmniBridgeTx: transformMetaTransaction(approvalOmniBridgeTx),
+      relayTestFundsToOmniBridgeTx: transformMetaTransaction(
+        relayTestFundsToOmniBridgeTx,
+      ),
+      transferCowTokenToCowDao: transformMetaTransaction(
+        transferCowTokenToCowDao,
+      ),
+    },
```

Comment:
I personally preferred the array version, as now any information on the order is lost, while getting back what each of the transactions is supposed to represent.
I know that there is a function that orders them, but still the function `generateProposal` does not really generate the proposal.

Reply:
I personally prefer to the strongly typed information. Together with the toArray helper, it has strictly more information and the single tx are easier to access. This will be handy akso in future PRs for testing the ArbitrayMessageBridge Tx.

If you want, we can have two functions `generateProposal` = `generateProposalAsIndexedTx` * `toArray`

Reply:
I like the proposal of renaming this function and then combining it with the array generation to transform it into `generateProposal`. :heavy_check_mark: 

Also, note that you could write this block of code as:
```
  const steps: Record<typeof steps[number], JsonMetaTransaction> =
    Object.fromEntries(
      Object.entries({
        cowDaoCreationTransaction,
        teamControllerCreationTransaction,
        investorFundsTargetCreationTransaction,
        cowTokenCreationTransaction,
        virtualCowTokenCreationTransaction,
        approvalOmniBridgeTx,
        relayTestFundsToOmniBridgeTx,
        transferCowTokenToCowDao,
      }).map(([key, tx]) => [key, transformMetaTransaction(tx)]),
    ) as Record<typeof steps[number], JsonMetaTransaction>;
```